gardener/container.go

```diff
@@ -145,16 +149,84 @@ func (c *container) Metrics() (garden.Metrics, error) {
 		}
 	}
 
+	networkStat, err := c.containerNetworkStat()
+	if err != nil {
+		return garden.Metrics{}, fmt.Errorf("could not read container network statistics, %w", err)
+	}
+
 	return garden.Metrics{
 		CPUStat:        actualContainerMetrics.CPU,
 		MemoryStat:     actualContainerMetrics.Memory,
 		DiskStat:       diskMetrics,
 		PidStat:        actualContainerMetrics.Pid,
 		Age:            actualContainerMetrics.Age,
 		CPUEntitlement: actualContainerMetrics.CPUEntitlement,
+		NetworkStat:    networkStat,
 	}, nil
 }
 
+func (c *container) containerNetworkStat() (garden.ContainerNetworkStat, error) {
+	ifName, err := c.Property(ContainerInterfaceKey)
+	if err != nil {
+		return garden.ContainerNetworkStat{}, err
+	}
+
+	stdout := new(bytes.Buffer)
+	stderr := new(bytes.Buffer)
+
+	process, err := c.Run(garden.ProcessSpec{
+		Path: "cat",
+		Args: []string{
+			networkStatPath(ifName, "rx_bytes"),
+			networkStatPath(ifName, "tx_bytes"),
+		},
+	}, garden.ProcessIO{
+		Stdout: stdout,
+		Stderr: stderr,
+	})
+
+	if err != nil {
+		return garden.ContainerNetworkStat{}, fmt.Errorf("running process failed, %w", err)
+	}
+
+	exitStatus, err := process.Wait()
+	if err != nil {
+		return garden.ContainerNetworkStat{}, err
+	}
+
+	if exitStatus != 0 {
+		return garden.ContainerNetworkStat{}, fmt.Errorf("running process failed with exit status %d, error %q", exitStatus, stderr.String())
+	}
+
+	stats := strings.Split(strings.TrimSpace(stdout.String()), "\n")
+	if len(stats) != 2 {
+		return garden.ContainerNetworkStat{}, fmt.Errorf("expected two values but got %q", stdout.String())
+	}
+
+	for idx, s := range stats {
+		stats[idx] = strings.TrimSpace(s)
+	}
+
+	rxBytes, err := strconv.ParseUint(stats[0], 10, 64)
+	if err != nil {
+		return garden.ContainerNetworkStat{}, fmt.Errorf("could not parse rx_bytes value %q, %w", stats[0], err)
+	}
+
+	txBytes, err := strconv.ParseUint(stats[1], 10, 64)
+	if err != nil {
+		return garden.ContainerNetworkStat{}, fmt.Errorf("could not parse tx_bytes value %q, %w", stats[1], err)
+	}
+
+	return garden.ContainerNetworkStat{
+		RxBytes: rxBytes,
+		TxBytes: txBytes,
+	}, nil
+}
+
+func networkStatPath(ifName, stat string) string {
+	return filepath.Join("/sys/class/net", ifName, "statistics", stat)
```

Comment:
This will likely fail on windows containers. We should make sure to test this as well.

Reply:
Hi @geofffranks,

we now added a noop implementation that is used on windows and in case the feature flag, [introduced here](https://github.com/cloudfoundry/garden-runc-release/pull/294), is disabled.

e87a8d4bb577828795ea3379718939a98b32c175